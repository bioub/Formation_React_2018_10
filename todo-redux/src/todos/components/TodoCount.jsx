import React, { Component } from 'react';
import { connect } from 'react-redux';
import { selectTodosCount } from '../selectors';

/*
function TodoCount({count = 0}) {
  return <div>{count} todos</div>
}
*/

export class TodoCount extends Component {
  render() {
    const {count = 0} = this.props;
    return <p>{count} todos</p>
  }
}

/*
TodoCount.propTypes = {

}
*/

const mapStateToProps = (state) => ({
  // exercice, sans ajouter de nouvelle clé au state
  // créer un selecteur qui récupère du state
  // le nombre d'élément du tableau todos
  count: selectTodosCount(state),
});

export default connect(mapStateToProps)(TodoCount);