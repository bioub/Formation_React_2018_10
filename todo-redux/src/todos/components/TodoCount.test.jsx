import React from 'react';
import ReactDOM from 'react-dom';
import { TodoCount } from './TodoCount';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TodoCount count={10} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot equals previous snapshot', () => {
  const tree = renderer
    .create(<TodoCount count={10} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

