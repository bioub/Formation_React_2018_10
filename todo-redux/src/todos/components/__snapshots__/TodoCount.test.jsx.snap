// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot equals previous snapshot 1`] = `
<p>
  10
   todos
</p>
`;
